provide/Services/NatsClient.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NATS.Client;
+
+namespace provide 
+{
+    public class NatsClient 
+    {
+        private string host;
+        private string path;
+        private string scheme;
+        private string token;
+        private readonly string natsUrl;
+        private IConnection connection;
+        private Dictionary<string, IAsyncSubscription> subscriptions = new Dictionary<string, IAsyncSubscription>();
+
+        public NatsClient(string token)
+        {
+            this.token = token;
+        }
+
+        public NatsClient(string host, string path, string scheme, string token)
+        {
+            this.host = host;
+            this.path = path;
+            this.scheme = scheme;
+            this.token = token;
+            this.natsUrl = $"{scheme}://${host}/${path}/";
+        }
+
+        public void Connect()
+        {
+            var opts = ConnectionFactory.GetDefaultOptions();
+            SetJWTAndNKeyHandlers(opts);
+            // if not provided it will use default local server url
+            opts.Url = this.natsUrl;
+            // add try catch?
+            this.connection = new ConnectionFactory().CreateConnection(opts);
```

Comment:
should we add some logging or leave ex handling to PBBP?

Reply:
maybe throw a nats exception, ismilar to the one bellow? throw new NATSBadSubscriptionException

Reply:
Yeah, think try/catch pattern would be good here.

Also, who has to handle reconnecting if the call to `Connect()` fails? What happens if a connection goes away after some time? Would be good to automatically reestablish it. Is there a way the caller can do that?

Reply:
Regarding reconnect, we are using default opts here, and 2 reconnect related fields are set there by default:
MaxReconnect = 60 // Gets or sets the maxmimum number of times a connection will attempt to reconnect.
ReconnectWait = 2000 //  Gets or sets the amount of time, in milliseconds, the client will wait before attempting a reconnection